dvc/external_repo.py

```diff
@@ -195,11 +195,26 @@ def get_external(self, path, dest):
 
         self._check_repo(path_info, metadata.repo)
         if metadata.output_exists:
+            import dvc.prompt as prompt
+            from dvc.exceptions import ConfirmRemoveError
+
+            dest_path_info = PathInfo(dest)
+            force = False
+            if self.tree.exists(dest_path_info):
+                msg = (
+                    "file '{}' already exists and it is going to be replaced."
+                    " Are you sure you want to proceed?".format(str(path_info))
+                )
+
+                if not prompt.confirm(msg):
+                    raise ConfirmRemoveError(str(path_info))
+                force = True
+
             repo = metadata.repo
             cache = repo.cache.local
             # fetch DVC and git files to tmpdir cache, then checkout
             save_info = self._fetch_to_cache(path_info, repo, None)
-            cache.checkout(PathInfo(dest), save_info)
+            cache.checkout(dest_path_info, save_info, force=force)
```

Comment:
Hi @julsal !

Thanks for the PR!

By default `force=False`, so if the output already exists and it differs from the desired one, then we will prompt user for removal confirmation. E.g.
```
#!/bin/bash                                                                       
                                                                                  
set -e                                                                            
set -x                                                                            
                                                                                  
rm -rf mytest                                                                     
mkdir mytest                                                                      
cd mytest                                                                         
                                                                                  
echo data > data                                                                  
dvc get https://github.com/iterative/dataset-registry get-started/data.xml -o data # will prompt
```
So this whole change is not really needed :slightly_frowning_face: You probably got confused by running `dvc get ...` twice, in which case it won't prompt you simply because the files are identical, so it doesn't have to prompt.

Reply:
Hi @efiop,

Thanks for the quick reply.

The change comes in the context of #3673, which shows that dvc asks for confirmation only after downloading the file. For this reason I introduced an earlier prompt. `force=True` comes to avoid another prompt.

Does it make sense when considering the issue?

Reply:
@julsal Ah, I see. Indeed, there was a misunderstanding on my side. Thank you for pointing out!

Just to clarify, this is a good PR :+1: and counts as a pass for the application purposes, but we want to reconsider from the tree arch perspective, which will require some more effort. So just to not waste your more of your time on this, we could close the PR for now.

Reply:
@efiop I agree. A future deeper refactoring might find a more elegant solution. Nonetheless, if you find useful, I changed the PR to keep only the test cases for file replacement.

Reply:
Thank you @julsal  :pray: